src/cascadia/WinRTUtils/inc/WtExeUtils.h

```diff
@@ -57,37 +74,35 @@ _TIL_INLINEPREFIX std::wstring GetWtExePath()
         // `wt.exe` on the %PATH% is us or not. Fortunately, _our_ execution alias
         // is located in "%LOCALAPPDATA%\Microsoft\WindowsApps\<our package family
         // name>", _always_, so we can use that to look up the exe easier.
-        try
+        if (IsPackaged())
         {
-            const auto package{ winrt::Windows::ApplicationModel::Package::Current() };
-            const auto id = package.Id();
-            const std::wstring pfn{ id.FamilyName() };
-            if (!pfn.empty())
+            try
             {
-                const std::filesystem::path windowsAppsPath{ wil::ExpandEnvironmentStringsW<std::wstring>(LocalAppDataAppsPath.data()) };
-                const std::filesystem::path wtPath = windowsAppsPath / pfn / (IsDevBuild() ? WtdExe : WtExe);
-                return wtPath;
+                const auto package = winrt::Windows::ApplicationModel::Package::Current();
+                const auto id = package.Id();
+                const auto pfn = id.FamilyName();
+                if (!pfn.empty())
+                {
+                    const std::filesystem::path windowsAppsPath{ wil::ExpandEnvironmentStringsW<std::wstring>(LocalAppDataAppsPath.data()) };
+                    const std::filesystem::path wtPath = windowsAppsPath / std::wstring_view{ pfn } / (IsDevBuild() ? WtdExe : WtExe);
+                    return wtPath;
+                }
             }
+            CATCH_LOG();
         }
-        CATCH_LOG();
 
         // If we're here, then we couldn't resolve our exe from the package. This
         // means we're running unpackaged. We should just use the
         // WindowsTerminal.exe that's sitting in the directory next to us.
         try
         {
-            HMODULE hModule = GetModuleHandle(nullptr);
-            THROW_LAST_ERROR_IF(hModule == nullptr);
-            std::wstring dllPathString;
-            THROW_IF_FAILED(wil::GetModuleFileNameW(hModule, dllPathString));
-            const std::filesystem::path dllPath{ dllPathString };
-            const std::filesystem::path rootDir = dllPath.parent_path();
-            std::filesystem::path wtPath = rootDir / WindowsTerminalExe;
-            return wtPath;
+            std::filesystem::path module = wil::GetModuleFileNameW<std::wstring>(nullptr);
+            module.replace_filename(WindowsTerminalExe);
+            return module;
         }
         CATCH_LOG();
 
-        return L"wt.exe";
+        return std::wstring{ WtExe };

```

Comment:
This was really confusing for me... It just wouldn't compile without explicitly specifying the type again, because it was complaining about the constructor being explicit.

Reply:
Yep. Have seen this before. Never figured out why.